tests/test_start_date.py

```diff
@@ -0,0 +1,142 @@
+            with self.subTest(stream=stream_name):
+
+                replication_method = stream_metadata.get(self.REPLICATION_METHOD)
+
+                first_sync_count = first_sync_record_count_by_stream.get(stream_name,0)
+                second_sync_count = second_sync_record_count_by_stream.get(stream_name,0)
+
+                self.first_sync_records = self.filter_output_file_for_records(
+                    first_sync_all_records_by_stream,
+                    stream_name
+                )
+
+                self.second_sync_records = self.filter_output_file_for_records(
+                    second_sync_all_records_by_stream,
+                    stream_name
+                )
+
+                if  replication_method == self.FULL_TABLE:
+                    """
+                    Testing Criteria:
+                    1. Verify that Sync B includes the same number of records as Sync A
+                    2. Verify the saved state following each sync does not contain a bookmark value
+                       for the stream.
+                    3. Verify that all records in Sync B are included in Sync A.
+                    """
+                    # Criteria 1
+                    self.assertEqual(first_sync_count, second_sync_count)
+
+                    # Criteria 2
+                    self.assertNotIn(stream_name, first_sync_state['bookmarks'])
+                    self.assertNotIn(stream_name, second_sync_state['bookmarks'])
+
+                    # Criteria 3
+                    first_sync_unique_records = self.get_unique_records(stream_name,
+                                                                        self.first_sync_records)
+                    second_sync_unique_records = self.get_unique_records(stream_name,
+                                                                         self.second_sync_records)
+                    self.assertSetEqual(first_sync_unique_records, second_sync_unique_records)
+                else:
+                    """
+                    Testing Criteria:
+                    1. Verify the saved state following each sync contains an appropriate bookmark
+                       value that corresponds to the stream’s replication key.
+                    2. Verify the record count for each stream is greater in Sync A than Sync B.
+                    3. Verify all records in Sync A and Sync B have replication key values which are
+                       greater than or equal to the corresponding start date for that sync.
+                    """
+                    # Criteria 1
+                    self.assertGreaterEqual(first_sync_count, second_sync_count)
+
+                    # Criteria 2
+                    self.assertIn(stream_name, first_sync_state['bookmarks'])
+                    self.assertIn(stream_name, second_sync_state['bookmarks'])
+
+                    # Criteria 3
+                    self.verify_replication_key_values(stream_name)
```

Comment:
I don't see the value in moving this into the methods above. This make the test less readable, I have to now go from method to method to see what the assertion is actually doing.